core/src/main/java/org/jruby/ir/instructions/YieldInstr.java

```diff
@@ -17,15 +17,17 @@
 import org.jruby.runtime.ThreadContext;
 import org.jruby.runtime.builtin.IRubyObject;
 
-public class YieldInstr extends TwoOperandResultBaseInstr implements FixedArityInstr {
+public class YieldInstr extends TwoOperandResultBaseInstr implements FixedArityInstr, Site {
     public final boolean unwrapArray;
+    private long callSiteId;
 
     public YieldInstr(Variable result, Operand block, Operand arg, boolean unwrapArray) {
         super(Operation.YIELD, result, block, arg == null ? UndefinedValue.UNDEFINED : arg);
 
         assert result != null: "YieldInstr result is null";
 
         this.unwrapArray = unwrapArray;
+        this.callSiteId = CallBase.callSiteCounter++;
```

Comment:
it doesn't matter if some end up with the same id (under concurrent execution), right?

Reply:
For today, this is not an issue because the conflict needs to be within the same scope we are deciding to inline on.  So it is more a tuple of {IRScope, callsiteId} as identity of the callsite we are interested in.  We cannot parallel build the same scope and we also have a limitation on multiple inlinings happening.  When we do support multiple inlinings we should revisit this (this scenario I suspect would be rare as well as both scopes would need to inline together then inline again -- 3 inlines where 2 methods were built at same moment).  I did not add this originally so this is largely just a continuation of what we do in CallBase.  You do make me realize when we get to recursive inlining we need to be aware we will have potentially n identical calls.  Even with atomic updates we need to cope with uniqueness.